backend/Routes/Controllers/auth/index.js

```diff
@@ -0,0 +1,282 @@
+const auth = require('../../../Models/auth');
+const middleware = require('../auth/middlewares');
+const ajv = require('../../../Schema');
+const express = require('express');
+const router = express.Router();
+
+const {
+  signupSchema,
+  loginSchema,
+  updateUserSchema,
+  updatePasswordSchema,
+  forgotPasswordSchema,
+  resetPasswordSchema,
+  verifyEmailSchema,
+} = require('../../../Schema/auth');
+
+/**
+ *
+ * @param {Array} errArray
+ * @return {String}
+ */
+function sumErrors(errArray) {
+  const cb = (a, b) => a + b.message + ', ';
+  return errArray.reduce(cb, '');
+}
+
+router.post('/signup', async (req, res) => {
+  let validate = ajv.compile(signupSchema);
+  let valid = validate(req.body);
+  if (!valid) {
+    return res.status(400).json({
+      success: false,
+      error: sumErrors(validate.errors),
+      results: null,
+    });
+  }
+  try {
+    let results = await auth.signup(req.body);
```

Comment:
Instead of using async await here, how about using promise only?
Code will be shorter and easy to read.
```js
auth.signup(req.body)
.then(results=>res.json())
.catch(error=>res.json())
```

Reply:
and everywhere else too.

Reply:
ok